[ZBC2014] New build recipe for ZBC2014

guestdaniel opened the pull request:
Added build_tarball.jl for Zilany, Bruce, and Carney (2014) auditory-nerve model C source code. The script builds `libzbc2014`, a shared library for Julia packages to provide access to the model code.

giordano commented on Z/zbc2014/build_tarballs.jl:
```diff
@@ -0,0 +1,27 @@
+${CC} -c -fPIC model_IHC.c -o model_IHC.o
+${CC} -c -fPIC model_Synapse.c -o model_Synapse.o
+${CC} -shared -o "libzbc2014.${dlext}" model_IHC.o model_Synapse.o complex.o
+mkdir ${libdir}
+cp "libzbc2014.${dlext}" ${libdir}
```
```suggestion
install -Dvm 755 "libzbc2014.${dlext}" -t "${libdir}"
```

giordano commented on Z/zbc2014/build_tarballs.jl:
```diff
@@ -0,0 +1,27 @@
+    LibraryProduct("libzbc2014", :libzbc2014)
+]
+
+dependencies = []
```
```suggestion
dependencies = Dependency[]
```

giordano reviewed (changes requested):
At a quick glance I don't see any license in https://www.urmc.rochester.edu/MediaLibraries/URMCMedia/labs/carney-lab/codes/Zilany-2014-Code-and-paper.zip, which would mean we aren't allowed to build and redistribute the code.

guestdaniel commented:
> At a quick glance I don't see any license in https://www.urmc.rochester.edu/MediaLibraries/URMCMedia/labs/carney-lab/codes/Zilany-2014-Code-and-paper.zip, which would mean we aren't allowed to build and redistribute the code.

I have (informal) written consent from the original contributors to release the updated source files (i.e., the files located at https://github.com/guestdaniel/ZBC2014.jl_CSource) under the GPL license. Not sure if it suffices to state something along those lines in the source repo to clarify things, or if an additional level of formalism is needed to make the updated code redistributable? Please advise, and if needed I can consult with the other contributors to resolve!

giordano commented on Z/zbc2014/build_tarballs.jl:
```diff
@@ -0,0 +1,26 @@
+using BinaryBuilder
+
+name = "libzbc2014"
```
The name of the directory must match exactly this name.  Either change this or move the directory as appropriate.  See also https://docs.binarybuilder.org/stable/building/#Name

giordano reviewed (commented):
> I have (informal) written consent from the original contributors to release the updated source files (i.e., the files located at [guestdaniel/ZBC2014.jl_CSource](https://github.com/guestdaniel/ZBC2014.jl_CSource)) under the GPL license.

It'd be good to have a written record of that, for example in the README of your repo, quoting verbatim the communication you received from the copyright holder to distribute the code under the terms of the GPL license.

giordano commented on Z/zbc2014/build_tarballs.jl:
```diff
@@ -0,0 +1,26 @@
+platforms = supported_platforms()
+
+products = [
+    LibraryProduct("zbc2014", :zbc2014)
```
The produce name was fine as `lib...`

guestdaniel replied:
Ah, I think I understand now :) Made another small commit to revert the product name back to `libzbc2014` but match the `name` variable to the directory name.

guestdaniel commented:
@giordano Thanks for your assistance thus far!

I think I addressed the issues related to naming; let me know if there's anything else needed there...

I'll work on getting a copy of written consent for the license change included in the source-code repo README — currently everything is embedded in a long, winding email chain from a while back, so I will contact everyone again to get a "clean" document to put in the README. May take a while... will ping once that's ready!

stemann commented on Z/zbc2014/build_tarballs.jl:
```diff
@@ -0,0 +1,26 @@
+using BinaryBuilder
+
+name = "zbc2014"
```
Though the name is sufficiently random, that nobody would likely ever need this for anything more general, `zbc2014` still seems like an overly terse name for something very specific (the implementation for a scientific publication in a certain academic field).

```suggestion
name = "ZilanyBruceCarney2014"
```

Notably, to match https://github.com/guestdaniel/ZilanyBruceCarney2014.jl

guestdaniel replied:
I have no strong preference regarding the name being abbreviated or not. Happy to do what is advised. My only hesitation is that I glanced at a few of the other folders in Yggdrasil and did not any other packages using last names. Let me know what is advised here.

guestdaniel commented:
@giordano I've updated the source repository to include a cleaned up copy of an email chain between the original authors and myself; everyone is okay with licensing the modified code under GNU AGPL. Let me know if any additional steps remain.

giordano commented:
> I've updated the source repository to include a cleaned up copy of an email chain between the original authors and myself

Can you please point me to it?  I may be missing it.

guestdaniel commented:
Here you go: https://github.com/guestdaniel/ZBC2014.jl_CSource/blob/main/license_change.txt

guestdaniel commented:
@giordano Pinging again in case you have not seen the link I included above yet!

giordano reviewed (approved):
Yes, sorry, thanks for the reminder!
